erpnext/public/js/utils.js

```diff
@@ -590,6 +591,25 @@ erpnext.utils.map_current_doc = function(opts) {
 	}
 }
 
+erpnext.utils.clear_duplicates = function() {
+	const unique_items = new Map();
+	/*
+		Create a Map of items with
+		item_code => [qty, warehouse, batch_no]
+	*/
+	let items = [];
+
+	for (let item of cur_frm.doc.items) {
+		if (!(unique_items.has(item.item_code) && unique_items.get(item.item_code)[0] === item.qty &&
+			unique_items.get(item.item_code)[1] === item.warehouse && unique_items.get(item.item_code)[2] === item.batch_no)) {
+
+			unique_items.set(item.item_code, [item.qty, item.warehouse, item.batch_no]);
```

Comment:
in case of PO/SO `delivery_date`, `required_date` is also unique

Reply:
Added `delivery_date` and `required_date` check